esphome/components/scd30/sensor.py

```diff
@@ -47,7 +42,7 @@ def remove_altitude_suffix(value):
             ),
             cv.Optional(CONF_AUTOMATIC_SELF_CALIBRATION, default=True): cv.boolean,
             cv.Optional(CONF_ALTITUDE_COMPENSATION): cv.All(
-                remove_altitude_suffix,
+                cv.float_with_unit("altitude", "(m|m a.s.l.|m MAMSL|m MAML)"),
```

Comment:
`MAML` was not valid in the original regex.

Also does `MAMSL` and `MASL` need the `m` at the start when the first `M` is meters? (not sure why the original regex allowed that)

```suggestion
                cv.float_with_unit("altitude", "(m|m a.s.l.|m MAMSL|m MASL)"),
```

Reply:
Right!